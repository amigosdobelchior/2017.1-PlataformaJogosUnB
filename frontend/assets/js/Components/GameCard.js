import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom'
import { Card } from 'semantic-ui-react'
import {Grid, Row, Column} from 'react-cellblock';

const styleImage = {height: '100%'};
export default class GameCard extends React.Component{
  constructor(props){
    super(props);
    this.state = { games: [] };
  }

  loadGameFromServer(){
        fetch("/api/list/",
              {
                headers: new Headers({ "Content-Type": "application/json", "Accept": "application/json"}),
                method: "GET",
            })
        .then((response) => {
             return response.json();
            })
        .then(((games) => {
            this.setState({ games: games });
        }).bind(this))
        .catch((error) => {
            console.error(error);
        });
  }

  componentDidMount() {
        this.loadGameFromServer();
    }

  render(){
    return (

          <Card>
          <img src={this.props.data.card_image} style={styleImage}/>
            <Card.Content>
              <Card.Header>
                {this.props.data.name}
              </Card.Header>
            </Card.Content>
            <Card.Content extra>
              <img src={this.props.data.packages[0].platforms[0].icon} height="42" width="42" />
            </Card.Content>
          </Card>

      )
    }
}
