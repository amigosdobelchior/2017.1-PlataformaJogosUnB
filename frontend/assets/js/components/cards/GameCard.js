import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom'
import { Card } from 'semantic-ui-react'

export default class GameCard extends React.Component{
  constructor(props){
    super(props);
    this.state = { games: [] };
  }

  loadGameFromServer(){
        fetch("/api/list/",
              {
                headers: new Headers({ "Content-Type": "application/json", "Accept": "application/json"}),
                method: "GET",
            })
        .then((response) => {
             return response.json();
            })
        .then(((games) => {
            this.setState({ games: games });
        }).bind(this))
        .catch((error) => {
            console.error(error);
        });
  }

  componentDidMount() {
        this.loadGameFromServer();
    }

  render(){
    return (

          <Card>
          <img src={this.props.data.cover_image}  height="142"/>
            <Card.Content>
              <Card.Header>
                {this.props.data.name}
              </Card.Header>
            </Card.Content>
            <Card.Content extra>
            </Card.Content>
          </Card>

      )
    }
}
