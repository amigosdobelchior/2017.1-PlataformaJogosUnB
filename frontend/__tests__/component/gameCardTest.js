import React from 'react';
import GameCard from '../../assets/js/components/cards/GameCard';
import {Card, Image} from "semantic-ui-react";
import renderer from 'react-test-renderer';

test('Test render GameCard', () => {
  const component = renderer.create(
      <Card>
          <div style= 'lolo'>
            <Image src="./image.jpg" style= 'lala' />
          </div>
          <Card.Content>
              <Card.Header>
                  "nome"
              </Card.Header>
          </Card.Content>
          <Card.Content extra>
                  <img key='1' src="./image2.jpg" width='20' height='20' /> )
          </Card.Content>
      </Card>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
