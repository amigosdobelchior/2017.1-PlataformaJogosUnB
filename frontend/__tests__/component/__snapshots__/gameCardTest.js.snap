// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render GameCard 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <div
    style="lolo"
  >
    <img
      alt={undefined}
      className="ui image"
      height={undefined}
      src="./image.jpg"
      style="lala"
      width={undefined}
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      nome
    </div>
  </div>
  <div
    className="extra content"
  >
    <img
      height="20"
      src="./image2.jpg"
      width="20"
    />
     )
  </div>
</div>
`;
